### ISSUE TITLE
SCC fails to login

### ISSUE THREAD

**Issue by ROpdebee** [SiCKRAGETV/sickrage-issues#3392]
Can't search for anything on Sceneaccess anymore, seems to give an error related to timeouts while logging in (but it's perfectly reachable from the browser). Never gotten it to work since I added SCC, except for about a week ago but it only worked for a day and then this behavior started again.

```
SR Version:	BRANCH: (master) / COMMIT: (7045417b59bb23cbc2426e2e1f7b0e59f21a87fb)
Python Version:	2.7.10 (default, Aug 22 2015, 20:33:39) [GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.1)]
OS: Mac OS X 10.11
```
Log:
```
2015-10-19 18:35:30 ERROR    SEARCHQUEUE-DAILY-SEARCH :: [SceneAccess] :: Connection timed out (sockets) accessing getURL https://sceneaccess.eu/login Error: '__str__ returned non-string (type Error)'
Traceback (most recent call last):
  File "/Applications/SickRage/sickbeard/helpers.py", line 1609, in getURL
    resp = session.post(url, data=post_data, timeout=timeout, allow_redirects=True, verify=session.verify)
  File "/Applications/SickRage/lib/requests/sessions.py", line 507, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/Applications/SickRage/lib/requests/sessions.py", line 464, in request
    resp = self.send(prep, **send_kwargs)
  File "/Applications/SickRage/lib/requests/sessions.py", line 576, in send
    r = adapter.send(request, **kwargs)
  File "/Applications/SickRage/lib/cachecontrol/adapter.py", line 46, in send
    resp = super(CacheControlAdapter, self).send(request, **kw)
  File "/Applications/SickRage/lib/requests/adapters.py", line 370, in send
    timeout=timeout
  File "/Applications/SickRage/lib/requests/packages/urllib3/connectionpool.py", line 544, in urlopen
    body=body, headers=headers)
  File "/Applications/SickRage/lib/requests/packages/urllib3/connectionpool.py", line 344, in _make_request
    self._raise_timeout(err=e, url=url, timeout_value=conn.timeout)
  File "/Applications/SickRage/lib/requests/packages/urllib3/connectionpool.py", line 314, in _raise_timeout
    if 'timed out' in str(err) or 'did not complete (read)' in str(err):  # Python 2.6
TypeError: __str__ returned non-string (type Error)
2015-10-19 18:35:30 ERROR    SEARCHQUEUE-DAILY-SEARCH :: [SceneAccess] :: Unable to connect to provider
```

This is everything the log says (debug logging is enabled).

FYI, thing's I've tried: 
- [x] clearing the cache :x:
- [x] disabling SSL Certificate checks and cleared the cache :x:
- [x] changing to a different VPN server :x:
- [x] changing to a different VPN server and clearing the cache :x:

**Comment by SiCKRAGEB0T** [comment]
## Thanks for the bug/issue report
Please make sure to submit all of the [required information](https://github.com/SiCKRAGETV/sickrage-issues/blob/master/README.md), or we won't be able to help you!
Provide as **much** information as you can such as debug logs, commit hash, and tell us how to replicate the problem.For basic **questions** use the [ [forum] ](https://sickrage.tv/forums/), and for **Feature Requests** [ [Feathub] ](http://feathub.com/SiCKRAGETV/SickRage)
Also make sure the title accurately describes your issue. ("SR not working" etc. is not ok!)

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [ ] Post debug logs, either inline (for smaller logs) or using gist)

## Many issues can simply be solved by:
- Making sure you update to the latest version.
- Have you tried turning your device off and on again?
- Using the **search** function to see if this issue has already been reported/solved.
- Do a [GIT reset](https://github.com/SiCKRAGETV/sickrage-issues/wiki/FAQ%27s-and-Fixes#update-problems-try-this).
- Checking the [ [Wiki] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki) for [ [Guides] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Installation-&-Configuration-Guides), [ [FAQS] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/FAQ%27s-and-Fixes) [ [Main settings] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Settings-explained), [ [Show settings] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Show-settings-explained) [ [Remaining settings] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Remaining-settings-explained)
- Shutting down SickRage and removing the `cache/mako` & `cache/sessions` folders.
- If SickRage doesnt find any episodes you probably need to :
 - add a [Scene Exception](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Scene-exceptions-and-numbering)
 - Set the [Default Episode Status](https://github.com/SiCKRAGETV/sickrage-issues/wiki/FAQ%27s-and-Fixes#newly-aired-shows-are-not-downloading-and-set-to-skippedignored) to `WANTED`
 - Make sure you understand how [quality detection](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Quality-Settings#quality-detectiondetermination) works during searches.
 - Shutdown SickRage and remove the `cache.db` file.
- If you use a QNAP and have SSL errors try this [package](http://apps.qnap.community/11-community/17-qsickrage).
- If you have a `'ascii' codec can't encode character` error, than set your Locale to [UTF-8](https://github.com/SiCKRAGETV/sickrage-issues/wiki/FAQ%27s-and-Fixes#i-have-problems-with-special-characters-%C3%A9-etc-what-can-i-do)

*ps. If you enjoy Sickrage and like the support provided, than please consider making a [donation](https://github.com/SiCKRAGETV/SickRage/wiki/Donations) to keep SickRage alive.*

**Comment by neoatomic** [comment]
Try to :  Shutting down SickRage and removing the cache/mako & cache/sessions folders.

**Comment by ROpdebee** [comment]
By clearing the cache I meant removing everything in cache/ except images and cache.db
I didn't explicitly close SR when removing, I just removed and restarted it, but I tried that now and still the error persists.

**Comment by neoatomic** [comment]
Than I'm not sure what this is coursing. You pretty much try'd all the stuff i would requirement to troubleshoot. Not using any proxy etc.? 
(And double check if you are indeed running/linking python 2.7.10)

Sadly i cant test SCC as it is a closed provider, and i dont have an account...

**Comment by ROpdebee** [comment]
running `python -V` in the terminal gives 2.7.10 so yeah pretty sure it's the correct version. Sadly I don't have any invites left. I don't have any proxy running but just to run another test, I'll enable one and see what I get (you never know)

**Comment by ROpdebee** [comment]
Routing the requests through proxite.eu doesn't give an explicit error but it's not working either, can't find episodes when they're clearly there when manually searching in browser. Manually opening the proxified link gives `The requested resource could not be loaded. libcurl returned the error:
SSL connect error` on proxite.eu, so I'm guessing it's some kind of SSL issue?

**Comment by neoatomic** [comment]
That was my first idea indeed... However as you where running Python 2.7.10 that is not likely. 
But its connection related for sure.

**Comment by miigotu** [comment]
Some sites cannot be run through the built in proxies. If you are using a VPN why would you enable a proxy anyways -.-

**Comment by ROpdebee** [comment]
I'm not. I just tested to see if that maybe helped troubleshooting the issue. I haven't got a proxy enabled by default, I was just verifying that it wasn't the VPN for sure.

**Comment by miigotu** [comment]
@ROpdebee do you have non-ascii chars in your login?

**Comment by ROpdebee** [comment]
nope

**Comment by miigotu** [comment]
I did just fix a problem with RSS update in SCC, but login, search, and snatch all work fine for me even before fixing that.

https://github.com/SiCKRAGETV/SickRage/pull/2937

**Comment by ROpdebee** [comment]
Doubt that it's anything related, since this is happening on manual search as well. I've been looking around for other issues that look related and it seems that it's an SSL related issue, so I've tried the fixes listed [here](https://github.com/SiCKRAGETV/sickrage-issues/wiki/SSL-Errors) but to no avail, probably because I'm running OS X 10.11 El Capitan (I'm not an expert but I think that last part could be very true).

**Comment by miigotu** [comment]
@ROpdebee I suggest removing pyOpenSSL and making sure SR is using Python 2.7.10

**Comment by ROpdebee** [comment]
@miigotu Can't remove pyOpenSSL, it's a package installed in the system's python and I can't delete it because it's protected by SIP (System Integrity Protection), a new feature in El Capitan. Guessing I'm never getting rid of this error lol. oh well

**Comment by ROpdebee** [comment]
I managed to fix it, thought I was running OpenSSL 1.0.2 but apparently python was using 0.9.8 because SickRage was started using the system's python, which uses 0.9.8. I installed OpenSSL 1.0.2 through Homebrew, then reinstalled python through Homebrew and started SR with newly installed python and all works well.

Sorry for wasting your time @miigotu and @neoatomic, thanks for the help.
